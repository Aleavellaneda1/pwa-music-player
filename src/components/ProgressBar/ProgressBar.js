import React, { memo } from 'react';
import styles from './styles.scss';

export default memo((props) => {
  return <div className="progress-bar">
    <div className="progress-bar__mask" style={{ width: `${Math.floor(props.percent * 100)}%` }}>
      <div className="progress-bar__progress"></div>
    </div>
    <div className="progress-bar__bg"></div>
  </div>
});
