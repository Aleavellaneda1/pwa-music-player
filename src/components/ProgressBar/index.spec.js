import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './ProgressBar';
import { enzymeConfig } from '../../enzimeConfig';

enzymeConfig();

describe('ProgressBar', () => {
  it('renders defaultProps', () => {
    const component = renderer.create(<ProgressBar percent={1} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
