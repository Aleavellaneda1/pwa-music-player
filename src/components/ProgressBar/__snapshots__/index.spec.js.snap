// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders defaultProps 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__mask"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="progress-bar__progress"
    />
  </div>
  <div
    className="progress-bar__bg"
  />
</div>
`;
